Description: Patch to add --as-needed magic to ltmain.sh
 This patch needs to be applied after autoreconf is run, and autoreconf should
 be copying files, not symlinking (ie: the '-s' option should not be given) so
 that system-wide files aren't accidentally patched.
 .
 This should be more robust than copying a pre-patched version of ltmain.sh
 from debian/ on build, as ltmain.sh needs to be exactly the same version as 
 the rest of libtool.
Author: Christopher James Halse Rogers <raof@ubuntu.com>

--- libtool-2.2.6b/libltdl/config/ltmain.sh	2010-01-04 05:33:00.810981679 +0000
+++ libtool-as-needed/libltdl/config/ltmain.sh	2010-01-04 00:18:29.715388881 +0000
@@ -5041,6 +5041,11 @@
 	arg=$func_stripname_result
 	;;
 
+      -Wl,--as-needed)
+	deplibs="$deplibs $arg"
+	continue
+	;;
+
       -Wl,*)
 	func_stripname '-Wl,' '' "$arg"
 	args=$func_stripname_result
@@ -5394,6 +5394,15 @@
 	lib=
 	found=no
 	case $deplib in
+	-Wl,--as-needed)
+	  if test "$linkmode,$pass" = "prog,link"; then
+	    compile_deplibs="$deplib $compile_deplibs"
+	    finalize_deplibs="$deplib $finalize_deplibs"
+	  else
+	    deplibs="$deplib $deplibs"
+	  fi
+	  continue
+	  ;;
 	-mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe|-threads)
 	  if test "$linkmode,$pass" = "prog,link"; then
 	    compile_deplibs="$deplib $compile_deplibs"
